sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -226,19 +226,17 @@ def _convert_connection_string_to_kwargs(cls, conn_str, **kwargs):
 
         kwargs["fully_qualified_namespace"] = host
         kwargs["entity_name"] = entity_in_conn_str or entity_in_kwargs
-        # This has to be defined seperately to support sync vs async credentials.
-        kwargs["credential"] = cls._create_credential_from_connection_string_parameters(
-            token, token_expiry, policy, key
-        )
-        return kwargs
 
-    @classmethod
-    def _create_credential_from_connection_string_parameters(
-        cls, token, token_expiry, policy, key
-    ):
+        # Set the type to sync credentials, unless async credentials are passed in.
+        token_cred_type = kwargs.get("token_cred_type") or ServiceBusSASTokenCredential
+        key_cred_type = kwargs.get("key_cred_type") or ServiceBusSharedKeyCredential
```

Comment:
we could use `kwargs.pop(key_name, default_value)` to pop out the argument that was no longer necessary to the following method call.

```suggestion
        token_cred_type = kwargs.pop("token_cred_type", ServiceBusSASTokenCredential)
        key_cred_type = kwargs.pop("key_cred_type", ServiceBusSharedKeyCredential)
```